src/main/java/com/google/devtools/build/lib/skyframe/FilesystemValueChecker.java

```diff
@@ -453,7 +462,8 @@ private boolean artifactIsDirtyWithDirectSystemCalls(
       boolean trustRemoteValue =
           fileMetadata.getType() == FileStateType.NONEXISTENT
               && lastKnownData.isRemote()
-              && trustRemoteArtifacts;
+              && trustRemoteArtifacts
+              && lastKnownData.isRemotelyAvailable(epochMilli);
```

Comment:
As is, that looks like in the internal case, that will mark 100% of remote values as requiring rebuild, which would be unacceptable (unless we use Long.MAX_VALUE there). I would recommend running some tests against that change.

Reply:
I will update the internal code as well during import to make sure they behave the same as today.